src/useGlobalState.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+
+export function useGlobalState<S = any>(initialState?: S) {
+  const store: { state: S | undefined; setState: (state: S) => void; setters: any[] } = {
+    state: initialState,
+    setState(state: S) {
+      store.state = state;
+      store.setters.forEach(setter => setter(store.state));
+    },
+    setters: [],
+  };
+
+  return (): [S | undefined, (state: S) => void] => {
+    const [globalState, stateSetter] = useState<S | undefined>(store.state);
+
+    if (!store.setters.includes(stateSetter)) {
```

Comment:
Seems this `IfStatement` will be called on every re-render. Probably better to put it inside `useLayoutEffect`?

Reply:
yeah, i'll pull a new request